FileNotFoundException during :plugins:repository-hdfs:forbiddenApisJavaRestTest

Build scan: https://gradle-enterprise.elastic.co/s/u6j4allgydguu

Interesting part in the console log:
```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':plugins:repository-hdfs:forbiddenApisJavaRestTest'.
...
Caused by: org.gradle.internal.deprecation.DocumentedFailure$DocumentedExceptionWithCause: Cannot access input property 'classpath' of task ':plugins:repository-hdfs:forbiddenApisJavaRestTest'. Accessing unreadable inputs or outputs is not supported. Declare the task as untracked by using Task.doNotTrackState(). For more information, please refer to https://docs.gradle.org/8.4/userguide/incremental_build.html#disable-state-tracking in the Gradle documentation.
...
Caused by: java.io.UncheckedIOException: Failed to create MD5 hash for file '/opt/local-ssd/buildkite/builds/bk-agent-prod-gcp-1701176508246589041/elastic/elasticsearch-periodic-platform-support/test/fixtures/krb5kdc-fixture/testfixtures_shared/shared/hdfs/keytabs/elasticsearch.keytab' as it does not exist.
...
Caused by: java.io.FileNotFoundException: /opt/local-ssd/buildkite/builds/bk-agent-prod-gcp-1701176508246589041/elastic/elasticsearch-periodic-platform-support/test/fixtures/krb5kdc-fixture/testfixtures_shared/shared/hdfs/keytabs/elasticsearch.keytab (Permission denied)	
	at java.base/java.io.FileInputStream.open0(Native Method)	
	at java.base/java.io.FileInputStream.open(FileInputStream.java:216)	
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)	
	at org.gradle.internal.hash.DefaultFileHasher.hash(DefaultFileHasher.java:36)
```

--- a/plugins/repository-hdfs/build.gradle
+++ b/plugins/repository-hdfs/build.gradle
@@ -59,12 +59,6 @@ dependencies {
     runtimeOnly "org.slf4j:slf4j-nop:${versions.slf4j}"
     //  runtimeOnly("org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}") https://github.com/elastic/elasticsearch/issues/93714
 
-    // Set the keytab files in the classpath so that we can access them from test code without the security manager
-    // freaking out.
-    if (isEclipse == false) {
-        javaRestTestRuntimeOnly project(path: ':test:fixtures:krb5kdc-fixture', configuration:'krb5KeytabsHdfsDir')
-    }
-
     krb5Keytabs project(path: ':test:fixtures:krb5kdc-fixture', configuration: 'krb5KeytabsHdfsDir')
     krb5Config project(path: ':test:fixtures:krb5kdc-fixture', configuration: 'krb5ConfHdfsFile')
 }
@@ -187,7 +181,11 @@ for (int hadoopVersion = minTestedHadoopVersion; hadoopVersion <= maxTestedHadoo
                 }
             }
             testClassesDirs = sourceSets.javaRestTest.output.classesDirs
-            classpath = sourceSets.javaRestTest.runtimeClasspath + files(portsFileDir)
+            // Set the keytab files in the classpath so that we can access them from test code without the security manager
+            // freaking out.
+            classpath = sourceSets.javaRestTest.runtimeClasspath +
+              configurations.krb5Keytabs +
+              files(portsFileDir)
         }
     }
 

--- a/test/fixtures/krb5kdc-fixture/build.gradle
+++ b/test/fixtures/krb5kdc-fixture/build.gradle
@@ -52,6 +52,6 @@ artifacts {
     builtBy("postProcessFixture")
   }
   krb5KeytabsHdfsDir(file("$testFixturesDir/shared/hdfs/keytabs/")) {
-    builtBy("preProcessFixture")
+    builtBy("postProcessFixture")
   }
 }

